05/app.js

```diff
@@ -0,0 +1,126 @@
+﻿const namePattern = /^[A-Za-ząćęłńóśźżĄĘŁŃÓŚŹŻ\s]*$/;
+const zipPattern = /[0-9]{2}-[0-9]{3}$/;
+
+document.addEventListener('DOMContentLoaded', init);
+
+function init() {
+    
+    const formEl = document.querySelector('form');    
+
+    formEl.addEventListener('submit', validateForm);
+}
+
+function validateForm(e) {
+
+    const form = e.target;   
+    const errors = []
+
+    e.preventDefault();
+
+    resetErrors(form);
+
+    checkTextData(form,'firstName', errors);
+    checkTextData(form,'lastName', errors);
+    checkTextData(form,'street', errors);
+    checkTextData(form, 'zip', errors, zipPattern);
+    checkNumberData(form,'houseNumber', errors);
+    checkNumberData(form,'flatNumber', errors, false);
+    checkTextData(form,'city', errors);
+    checkSelectData(form,'voivodeship', errors);
+    
+    showErrors(form, errors);
+       
+}
+
+function showErrors(form, errors) {
+
+    const listErr = form.querySelector('.messages');
+    
+    if (errors.length > 0) {
+
+        errors.forEach(function(item){
+
+            const errItem = document.createElement('li');       
+            errItem.textContent = item.msg;
+            listErr.appendChild(errItem);
+
+            item.label.style = 'color : red';
+
+        });
+    } 
+    else {
+        
+        const errItem = document.createElement('li');       
+        errItem.textContent = 'Dane zostały wysłane prawidłowo';
+        listErr.appendChild(errItem);
+    }
+
+}
+
+function resetErrors(form) {
+
+    const label = form.querySelectorAll('label');
+    const listErr = form.querySelector('.messages');
+    
+    label.forEach(function(item){
+        item.style = 'color : black';
+    });
+    
+    listErr.textContent = '';
+}
+
+function checkTextData(form, name,  errors, pattern = namePattern) {
+        
+    const inputEl = form.elements[name];
```

Comment:
Lepiej byłoby przekazywać samą wartość - to rozwiązanie jest bardziej elastycznie bo nie wiążemy się z formularzem.